Move custom `segment` style from core to demo

ssorallen opened the pull request:
The `segment` styles are defined in layout.html in Polymer 0.5 and in
Polymer/layout for 0.8. Since the styles are only used here in the
core-icon demo, move them here to eliminate the overhead for other
clients of layout.html.

Commit:
Move custom `segment` style from core to demo

The `segment` styles are defined in layout.html in Polymer 0.5 and in
Polymer/layout for 0.8. Since the styles are only used here in the
core-icon demo, move them here to eliminate the overhead for other
clients of layout.html.

cdata commented on demo.html:
```diff
@@ -13,9 +13,20 @@
-  
+
   <style>
     segment {
```
Why not make this a valid element? I think `segment` was chosen arbitrarily.

ssorallen replied:
Sure thing.

cdata replied:
(`section` comes to mind, but `div` is fine too)

ssorallen replied:
These are tiny boxes, so I went for a class name instead of a `section`.

Commit:
Prefer a class name over invalid custom element

`segment` is not an existing HTML element and is not a valid custom
element name because it's missing a dash.

cdata commented:
LGTM :+1:
